controllers/operatorconfigmap_controller.go

```diff
@@ -654,31 +670,31 @@ func (c *OperatorConfigMapReconciler) reconcileSubscriptionValidatingWebhook() e
 }
 
 func (c *OperatorConfigMapReconciler) reconcileClientOperatorSubscriptionLabel() error {
-	subscriptionList := &opv1a1.SubscriptionList{}
-	err := c.List(c.ctx, subscriptionList, client.InNamespace(c.OperatorNamespace))
-	if err != nil {
-		return fmt.Errorf("failed to list subscriptions")
-	}
 
-	sub := utils.Find(subscriptionList.Items, func(sub *opv1a1.Subscription) bool {
-		return sub.Spec.Package == "ocs-client-operator"
-	})
+	subList := &opv1a1.SubscriptionList{}
+	if err := c.list(
+		subList,
+		client.MatchingFields{subPackageIndexName: "ocs-client-operator"},
+		client.InNamespace(c.OperatorNamespace),
+		client.Limit(1),
+	); err != nil {
+		return fmt.Errorf("failed to list subscriptions: %v", err)
+	}
 
-	if sub == nil {
-		return fmt.Errorf("failed to find subscription with ocs-client-operator package")
+	if len(subList.Items) != 1 {
+		return fmt.Errorf("failed to find ocs-client-operator subscription")
```

Comment:
Can we please differentiate between `0` and `> 1`. And provide a meaningful version in the log for each case?